### DISCUSSION TITLE
In Regards To Self Compiling & If My Steps To Completion Are Correct :)

### DISCUSSION THREAD

**Discussion by killlerpenguins** [Jyers/Marlin#253] [Q&A] [answered]
Please forgive me as I am new to all this and wanting to make sure I have a good understanding and that I am doing everything correctly. Just to sum up my setup and what I am wanting to accomplish. I have an Ender 3 V2 with both a BL Touch and Creality's Z-axis extension kit bringing my Z-axis height up from 250mm to 350mm. So with this setup I need a firmware configuration with both the BL Touch and the Z-axis height adjusted from 250mm to 350mm.

Below I will list the steps I am doing and starting at step 5 is where I am getting confused and have questions.

**Step 1:** Go to the resources section on Git
**Step 2:** Download the Source Code Zip.
**Step 3:** Unzip the zipped file.
**Step 4:** Launch VSCode and open the unzipped file via PlatformIO's Open Project.
(In the following steps This is how I am thinking it is done. Please correct me if and where I am wrong)
**Step 5:** Since I am needing the configuration files for the BL Touch and not wanting it with high speed, I would access the following file location. Configuration Files => E3V2 Templates => BLTouch-3x3.
**Step 6:** In the BLTouch-3x3 folder select both Config.h & config_adv.h files, right click them and select copy.
**Step 7:** Go to the Marlin section, right click and paste to replace the existing Config.h & config_adv.h files with the ones from the BLTouch-3x3 folder.
**Step 8:** Since I am needing to increase the Z-axis height from 250mm to 350mm, go into the config.h file, go to line 1175. Should read "#define Z_MAX_POS 250. Change 250 to 350.
**Step 9:** Save everything
A question I have before doing step 10 above (compiling the firmware) is there anything I need to delete and remove from the root folder or everything is fine left in the root folder and can continue with step 10 compiling the build?
**Step 10:** Compile Build with checkmark icon.
**Step 11:** Grab the newly created .bin file and put on empty SD card.
**Step 12:** Plug SD card into printer which should be powered off.
**Step 11:** Power printer on and printer should flash to the new firmware.
Once printer is fully flashed and at the menu screen, you can access the printers info section and see the updated firmware there.
**Step 12:** Once you have the firmware flashed, reset EEPROM and delete the .bin file from the SD card.
**Step 13:** Launch your slicer software as we need to now make a few changes to your machine.
**Step 14:** Change your build size Z-axis to 350mm to match what was changed in the firmware.
**Step 15:** In your Start G-Code add both M420 S1 & G29.

Are all the above steps correct and is there anything that needs adjusted and fixed?

One last thing in regards to changing the Z-axis to 350, once the firmware is flashed to the printer and if accessing the printers info, the display will still show the Z-axis being 250mm. What needs to also be changed in the project for it to properly reflect visually on the display when in the printers info section?

Thank you in advance for your guys help and support!

**Comment by Schnabulation** [comment, marked as answer]
I think you are pretty spot on actually. A couple of things I'd like to mention:

- Once you have the firmware flashed, reset EEPROM and delete the .bin file from the SD card.
- The bed size on the display is created from X_BED_SIZE x Y_BED_SIZE x Z_MAX_POS, so when you change that value it should show correctly.
- Make sure to change the build size in your slicer.
- Look through the config files - there are plenty of things you can change. However: the pre-configs from Jyers are a pretty good start point.

**Reply by killlerpenguins** [reply]
@Schnabulation  Shoot, I knew I was forgetting to add a few more steps. LOL Thank you for that. I updated my original discussion and added in 4 more steps (Steps 12-15) showing what you had mentioned as well as about adding both M420 S1 & G29 to your Start G-Code. 

My 2 main concerns for making sure I was correct was:
**1.** Replacing the config.h and config_adv.h files in the Marlin folder with the ones from the BLTouch-3x3 folder.
**2.** I didn't need to remove anything such as the Configuration Files folder or anything else before compiling the build.

**Making a small edit to my reply. In regards to it not showing the specified build size in the info section of the printer, I just compiled another new build and re-flashed the printer and it is now showing it in the info section. Def weird it didn't before.**

**Comment by Tw0bit** [comment]
Thank you @killlerpenguins. I have spent a few hours looking how to get PlatformIO to build the template.

**Reply by killlerpenguins** [reply]
@Tw0bit  You are welcome! Happy I could help!

**Comment by Cal-LF** [comment]
Your steps for compiling are a good instruction. My I use it to post on user group?

**Reply by killlerpenguins** [reply]
@Cal-LF  Sorry for just now seeing and responding to your message. If you have not already, be my guest! Happy they are clear and helpful! Thank you!
